Sources/Property.swift

```diff
@@ -678,3 +711,71 @@ public final class MutableProperty<Value>: MutablePropertyProtocol {
 		observer.sendCompleted()
 	}
 }
+
+extension MutableProperty {
+
+	/// Returns a `BindingTarget` that invokes the given setter to generate a
+	/// new value for this property.
+	///
+	/// Consider the following example:
+	///
+	/// ```
+	/// let personProperty = MutableProperty(Person(firstName: "Steve", lastName: "McQueen"))
+	/// let firstNameBinding = personProperty.bindingTarget {
+	///         return Person(firstName: $1, lastName: $0.lastName)
+	///     }
+	///
+	/// // Sets only the firstName value
+	/// firstNameBinding <~ SignalProducer(value: "Lightning")
+	///
+	/// // The personProperty now has a new Person value
+	/// ```
+	///
+	/// - note: The lifetime of this `BindingTarget` is tied to the underlying
+	///         `MutableProperty` instance.
+	///
+	/// - parameters:
+	///	  - setter: A closure that takes a value of type `U` to produce a new
+	///             value for this property
+	///
+	public func bindingTarget<U>(setter: @escaping (Value, U) -> Value) -> BindingTarget<U> {
```

Comment:
This is pretty interesting, but it seems like we should consider this in a separate PR. I don't see any relation or dependency between the two? (Other than these were both originally part of your lens concept.)